pkg/cloud/amazon/s3_storage.go

```diff
@@ -810,41 +850,38 @@ func (s *s3Storage) List(ctx context.Context, prefix, delim string, fn cloud.Lis
 		return err
 	}
 
-	var fnErr error
-	pageFn := func(page *s3.ListObjectsOutput, lastPage bool) bool {
-		for _, x := range page.CommonPrefixes {
-			if fnErr = fn(strings.TrimPrefix(*x.Prefix, dest)); fnErr != nil {
-				return false
-			}
-		}
-		for _, fileObject := range page.Contents {
-			if fnErr = fn(strings.TrimPrefix(*fileObject.Key, dest)); fnErr != nil {
-				return false
-			}
-		}
-
-		return true
-	}
-
-	var s3Input *s3.ListObjectsInput
+	var s3Input *s3.ListObjectsV2Input
```

Comment:
ListObjectsV2 is actually a different underlying API request, so there might be behaviour changes/3rd party incompatibilities we'll end up learning about here.